tensorflow/python/client/timeline.py

```diff
@@ -607,7 +617,48 @@ def _show_memory_counters(self):
                                         total_bytes)
     self._allocator_maximums = alloc_maxes
 
-  def analyze_step_stats(self, show_dataflow=True, show_memory=True):
+  def _preprocess_op_time(self, op_time):
+    """Modified the start and end of ops in step stats"""
+    if op_time == "schedule":
+      self._step_stats = self._origin_step_stats
+      return
+    self._step_stats = copy.deepcopy(self._origin_step_stats)
+    # Separate job task and gpu tracer stream
+    stream_all_stats = []
+    job_stats = []
+    for stats in self._step_stats.dev_stats:
+      if "/stream:all" in stats.device:
+        stream_all_stats.append(stats)
+      elif "/job" in stats.device:
+        job_stats.append(stats)
+
+    # Record the start time of the first kernel and the end time of
+    # the last kernel for all ops.
+    op_gpu_start = {}
+    op_gpu_end = {}
```

Comment:
How does this work for CPU only jobs?

Reply:
There will be no influence on CPU only jobs, because we only check ops that relate to the kernels profiled in "/stream:all", which are all of the gpu kernels. Also, the current profiling method on CPU only ops in tensorflow does not have the async problem, so it would be ok to leave them where they were.